src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala

```diff
@@ -72,11 +72,20 @@ object HashAlgoConstraint {
   /*
    * Actually do not hash the result
    */
-  object PLAIN  extends HashAlgoConstraint {
+  object PLAIN extends HashAlgoConstraint {
     override def hash(input:Array[Byte]) : String = new String(input, "UTF-8")
     override val prefix = "plain"
   }
 
+  object PreHashed extends HashAlgoConstraint {
+    override def hash(input:Array[Byte]) : String = new String(input, "UTF-8")
+    override val prefix = "pre-hashed"
+  }
+
+  object SCRIPT extends HashAlgoConstraint {
+    override def hash(input:Array[Byte]) : String = new String(input, "UTF-8")
+    override val prefix = "script"
```

Comment:
I'll let the evaluation do it for me. 

Not a problem to me if it does nothing but i can identify and help the user to enter a script